_api_app/app/Sites/Sections/SiteSectionsDataService.php

```diff
@@ -308,48 +308,59 @@ public function saveValueByPath($path, $value)
             $old_name = $sections['section'][$order]['name'];
             $old_title = isset($sections['section'][$order]['title']) ? $sections['section'][$order]['title'] : '';
             $new_name = $this->getUniqueSlug($old_name, $value);
+            $success=false;
 
             if (empty($value)) {
                 $ret['value'] = $old_title;
                 $ret['error_message'] = 'Section name cannot be empty!';
                 return $ret;
             }
 
-            // Rename section name
-            $this->setValueByPath(
-                $sections,
-                'section/' . $order . '/name',
-                $new_name
-            );
-
-            // Rename section background path
-            if (isset($sections['section'][$order]['mediafolder'])) {
-                $mediafolder = $new_name . '-background';
-
-                @rename(
-                    realpath($this->MEDIA_ROOT) . '/' . $sections['section'][$order]['mediafolder'],
-                    realpath($this->MEDIA_ROOT) . '/' . $mediafolder
-                );
-
+            //Compare the title when we rename it
+            if ($old_name===$new_name && $old_title!==$value){
```

Comment:
Please use spaces around the comparison operators - `$old_name===$new_name` -> `$old_name === $new_name`.
Try to follow the code style you see around.

Reply:
It seems to me that this `if` block is redundant. You could do the same check down on line `325`. And then do the `$ret` assignment in an `else` branch down below line `363`.